pkg/ovs/openflow/ofctrl_builder_test.go

```diff
@@ -15,30 +15,2148 @@
+				expectedVlanMask:   mask2,
+				expectedMatcherStr: "vlan_tci=0x0000/0x1000",
+			},
+			{
+				nonVLAN:          false,
+				vlanID:           1,
+				vlanMask:         &mask1,
+				expectedVlanMask: mask1,
+				expectedMatchField: &openflow15.MatchField{
+					Class:   openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					Field:   openflow15.OXM_FIELD_VLAN_VID,
+					HasMask: true,
+					Value:   &openflow15.VlanIdField{VlanId: 0x1001},
+					Mask:    &openflow15.VlanIdField{VlanId: 0x1fff},
+				},
+				expectedMatcherStr: "dl_vlan=1",
+			},
+			{
+				nonVLAN:          false,
+				vlanID:           1,
+				vlanMask:         nil, // When nonVLAN is false, if vlan mask is not set, vlan mask 0x1fff will be used by default.
+				expectedVlanMask: mask1,
+				expectedMatchField: &openflow15.MatchField{
+					Class:   openflow15.OXM_CLASS_OPENFLOW_BASIC,
+					Field:   openflow15.OXM_FIELD_VLAN_VID,
+					HasMask: true,
+					Value:   &openflow15.VlanIdField{VlanId: 0x1001},
+					Mask:    &openflow15.VlanIdField{VlanId: 0x1fff},
+				},
+				expectedMatcherStr: "dl_vlan=1",
+			},
+		}
+
+		for _, tc := range testCases {
+			fb := table.BuildFlow(1).MatchVLAN(tc.nonVLAN, tc.vlanID, tc.vlanMask).(*ofFlowBuilder)
+			assert.Equal(t, tc.nonVLAN, fb.Match.NonVlan)
+			assert.Equal(t, tc.vlanID, *fb.Match.VlanId)
+			assert.Equal(t, tc.expectedVlanMask, *fb.Match.VlanMask)
+
+			flowMod := getFlowMod(t, fb.Done())
+			assert.Equal(t, 1, len(flowMod.Match.Fields))
+			checkMatchField(t, tc.expectedMatchField, &flowMod.Match.Fields[0])
+			assert.Contains(t, getFlowModMatch(flowMod), tc.expectedMatcherStr)
+		}
+	})
+	t.Run("SetHardTimeout", func(t *testing.T) {
+		fb := table.BuildFlow(1).SetHardTimeout(uint16(3600)).(*ofFlowBuilder)
+		assert.Equal(t, uint16(3600), fb.ofFlow.HardTimeout)
+
+		flowMod := getFlowMod(t, fb.Done())
+		assert.Equal(t, uint16(3600), flowMod.HardTimeout)
+		assert.Contains(t, getFlowModBaseString(flowMod), "hard_timeout=3600")
+	})
+	t.Run("SetIdleTimeout", func(t *testing.T) {
+		fb := table.BuildFlow(1).SetIdleTimeout(uint16(3600)).(*ofFlowBuilder)
+		assert.Equal(t, uint16(3600), fb.ofFlow.IdleTimeout)
+
+		flowMod := getFlowMod(t, fb.Done())
+		assert.Equal(t, uint16(3600), flowMod.IdleTimeout)
+		assert.Contains(t, getFlowModBaseString(flowMod), "idle_timeout=3600")
+	})
+	t.Run("MatchXXReg", func(t *testing.T) {
+		testCases := []struct {
+			regID              int
+			data               []byte
+			expectedMatcherStr string
+			expectedMatchField *openflow15.MatchField
+		}{
+			{
+				regID:              0,
+				data:               []byte{0x12, 0x34, 0x56},
+				expectedMatcherStr: "xxreg0=0x123456",
+				expectedMatchField: &openflow15.MatchField{
+					Class: openflow15.OXM_CLASS_NXM_1,
+					Field: openflow15.NXM_NX_XXREG0,
+					Value: &openflow15.ByteArrayField{Data: []byte{0x12, 0x34, 0x56}, Length: 3},
+				},
+			},
+			{
```

Comment:
Just to test different values. Besides, this can be also to test if the match string is as expected. For example, the first case should be `xxreg0=0x123456`, not `xxreg0=0x00123456`